### ISSUE TITLE
Compiling SWMM v5.2 on Ubuntu 20.04

### ISSUE THREAD

**Issue by rarygit** [USEPA/Stormwater-Management-Model#82]
So far, two errors when building SWMM v5.2 on Ubuntu 20.04

(i) File: inlet.c
Lines 1570 & 1898:  qMax = HUGE 
Solution:   change to qMax = BIG

(ii) File: output.c
Lines 37ff:   #elif              // Other platforms
and multiple error messages associated with F_OFF

Thanks for any suggestions to solve (ii) errors: 
'#elif with no expression', and 
'unknown type name ‘F_OFF’
  
+++++++++++++++++++++++++++++++++++++++++++++++++++++++

For example:

[ 60%] Building C object src/solver/CMakeFiles/swmm5.dir/output.c.o

.../Stormwater-Management-Model-develop/src/solver/output.c:36:37: error: #elif with no expression
   36 | #elif             // Other platforms
                                          ^
.../Stormwater-Management-Model-develop/src/solver/output.c:62:8: error: unknown type name ‘F_OFF’
   62 | static F_OFF     IDStartPos;           // starting file position of ID names

.../Stormwater-Management-Model-develop/src/solver/output.c:63:8: error: unknown type name ‘F_OFF’
   63 | static F_OFF     InputStartPos;        // starting file position of input data
      |        ^~~~~

etc, etc

**Comment by LRossman** [comment]
To fix output.c change lines 36-37 to:
```
#else
    #define F_OFF off_t
```
Tested successfully on Ubuntu 16.04.

**Comment by rarygit** [comment]
@LRossman,  that works, brilliant! And many thanks for the quick solution!

I would like to test the run time with v5.2 compiled on Linux. 
Is there a test example on GitHub, relevant to the changes made in v5.2, which could be used for this purpose?

**Comment by LRossman** [comment]
A couple of things:

1. The major change made in v5.2 was the inclusion of street inlet modeling which doesn't have a test example on GitHub. There is a small example that comes with the EPA distribution package that can be found on the [EPA web site](https://www.epa.gov/water-research/storm-water-management-model-swmm).
2. If you want to compare dynamic wave routing run times (the most time consuming calculation made by SWMM) there are some larger sewer networks included in the Quality Assurance Report for Dynamic Wave Flow Routing that is also available from the EPA web site.
3. A bug was recently found in the controls.c module that affects run times for models with control rules. The fix is to replace the last section of code in the `updateActionList() `function with the following:
```
   // --- action not listed so add it to ActionList
    if ( !listItem )
    {
        listItem = (struct TActionList *) malloc(sizeof(struct TActionList));
        listItem->next = ActionList;
        ActionList = listItem;
    }
    listItem->action = a;
```

**Comment by rarygit** [comment]
@LRossman - thanks for replying!

I recompiled with your changes above and then ran the User Examples from your Quality Assurance Report, 2006.

I have attached the runtimes shown in the Terminal, as well as the Continuity errors and Warnings.
Run times ranged from 2 to 11 seconds.

[runtime_results_Ubuntu20.04_swmm.v5.2_develop.zip](https://github.com/USEPA/Stormwater-Management-Model/files/8203701/runtime_results_Ubuntu20.04_swmm.v5.2_develop.zip)


In all the inp files, I commented out NORMAL_FLOW_LIMITED  ON, in response to the error. 
Would this affect the run time?
What should 'NORMAL_FLOW_LIMITED' be set to?
